File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def getResult(obj, type1, type2):
    if obj[type1] > obj[type2]:
        return type1
    elif obj[type1] < obj[type2]:
        return type2
    else:
        return sorted([type1, type2])[0]

def solution(survey, choices):
    answer = ''
    
    obj = {
        "R": 0, "T": 0, "C": 0, "F": 0, "J": 0, "M": 0, "A": 0, "N": 0
    }
    
    for i in range(len(survey)):
        point = choices[i]
        if point == 1:
            obj[survey[i][0]] += 3
        elif point == 2:
            obj[survey[i][0]] += 2
        elif point == 3:
            obj[survey[i][0]] += 1
        elif point == 5:
            obj[survey[i][1]] += 1
        elif point == 6:
            obj[survey[i][1]] += 2
        elif point == 7:
            obj[survey[i][1]] += 3
    
    answer += getResult(obj, "R", "T")
    answer += getResult(obj, "C", "F")
    answer += getResult(obj, "J", "M")
    answer += getResult(obj, "A", "N")     
        
    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    const getPoint = (num) => {
        if (num === 1 || num === 7) {
            return 3;
        } else if (num === 2 || num === 6) {
            return 2;
        } else if (num === 3 || num === 5) {
            return 1;
        }
        return 0;
    }
    
    const ref = {
        "R": 0, "T": 0,
        "C": 0, "F": 0,
        "J": 0, "M": 0,
        "A": 0, "N": 0,
    }
    
    let answer = '';
    for (let i = 0; i < survey.length; i++) {
        const [firstItem, secondItem] = survey[i].split("");
        const choice = choices[i];
        const point = getPoint(choice);
        if (choice < 4) {
            ref[firstItem] += point;
        } else if (choice > 4) {
            ref[secondItem] += point;
        }  
    }
    
    if (ref["R"] >= ref["T"]) answer += "R";
    else answer += "T";
    if (ref["C"] >= ref["F"]) answer += "C";
    else answer += "F";
    if (ref["J"] >= ref["M"]) answer += "J";
    else answer += "M";
    if (ref["A"] >= ref["N"]) answer += "A";
    else answer += "N";
    
    return answer;
}